be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -421,6 +431,7 @@ Status SegmentWriter::append_block_with_partial_content(const vectorized::Block*
             }
             has_default_or_nullable = true;
             use_default_or_null_flag.emplace_back(true);
+            ++segment_pos;
```

Comment:
we'd better to avoid ++segment_pos in many places, it's very easy to miss some updates in some code branch.
You can record a segment_start_pos outside of this loop, and update the `segment_pos` at the beginning of this loop: `setment_pos = delta_pos + segment_start_pos`.